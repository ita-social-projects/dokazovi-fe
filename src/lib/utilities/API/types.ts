import { IDirection } from '../../types';

export type CreatePostRequestType = {
  title: string;
  content: string;
  directions: IDirection[];
  preview: string;
  type: {
    id: number;
  };
  previewImageUrl?: string;
  videoUrl?: string;
};

export type CreateTextPostRequestType = CreatePostRequestType;

export type CreateVideoPostRequestType = CreatePostRequestType & {
  videoUrl: string;
};

export type CreatePostRequestUnionType =
  | CreateTextPostRequestType
  | CreateVideoPostRequestType;

type UpdatePostRequestType = CreatePostRequestType & {
  id: number;
};

export type UpdateTextPostRequestType = UpdatePostRequestType;

export type UpdateVideoPostRequestType = UpdatePostRequestType & {
  videoUrl: string;
};

export type UpdatePostRequestUnionType =
  | UpdateTextPostRequestType
  | UpdateVideoPostRequestType;

export type PostResponseType = {
  author: {
    avatar: string;
    firstName: string;
    id: number;
    lastName: string;
    mainInstitution: {
      city: {
        id: number;
        name: string;
      };
      id: number;
      name: string;
    };
  };
  content: string;
  preview: string;
  postType: PostTypeResponseType;
  createdAt: string;
  directions: DirectionResponseType[];
  id: number;
  title: string;
  type: {
    id: number;
    name: string;
  };
  videoUrl?: string;
};

export type ExpertResponseType = {
  avatar: string;
  firstName: string;
  id: number;
  lastAddedPost: {
    id: number;
    title: string;
  };
  lastName: string;
  mainDirection: {
    id: number;
    name: string;
  };
  mainInstitution: {
    city: {
      id: number;
      name: string;
    };
    id: number;
    name: string;
  };
  qualification: string;
};

export type CreateTagRequestType = {
  tag: string;
};

export type TagResponseType = {
  id: number;
  tag: string;
};

export type VersionResponseType = {
  version: string;
};

export type DirectionResponseType = {
  id: number;
  name: string;
  label: string;
  color: string;
};

export type RegionResponseType = {
  id: number;
  name: string;
};

export type PostTypeResponseType = {
  id: number;
  name: string;
};

export type LoginResponseType = {
  accessToken: string;
  tokenType: string;
};

type GetResponseType<T> = {
  content: T[];
  empty: boolean;
  first: boolean;
  last: boolean;
  number: number;
  numberOfElements: number;
  size: number;
  sort: {
    empty: boolean;
    sorted: boolean;
    unsorted: boolean;
  };
  totalElements: number;
  totalPages: number;
};

export type ExpertsResponseType = GetResponseType<ExpertResponseType>;

export type PostsResponseType = GetResponseType<PostResponseType>;

export type GetTagsConfigType = {
  params: {
    value: string;
    limit?: number;
  };
};

type GetConfigType = {
  params: RequestParamsType;
};

export type GetExpertsConfigType = GetConfigType & {
  params: {
    directions?: number[];
    regions?: number[];
  };
};

export type GetPostsConfigType = GetConfigType;

export type RequestParamsType = {
  page?: number;
  size?: number;
  sort?: string[];
  directions?: number[];
  type?: number[];
  expert?: number;
  regions?: number[];
  tag?: number[];
};
