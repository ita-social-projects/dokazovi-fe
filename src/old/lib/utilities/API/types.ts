import { IDirection, IOrigin, IPost } from '../../types';

export type CreatePostRequestType = {
  title: string;
  authorsName?: string;
  authorsDetails?: string;
  content: string;
  directions: IDirection[];
  origins?: IOrigin[];
  preview: string;
  type: {
    id: number;
  };
  previewImageUrl?: string;
  videoUrl?: string;
  authorId?: number | null;
};

export type CreateTextPostRequestType = CreatePostRequestType;

export type CreateVideoPostRequestType = CreatePostRequestType & {
  videoUrl: string;
};

export type CreatePostRequestUnionType =
  | CreateTextPostRequestType
  | CreateVideoPostRequestType;

type UpdatePostRequestType = CreatePostRequestType & {
  id: number;
};

export type UpdateTextPostRequestType = UpdatePostRequestType;

export type UpdateVideoPostRequestType = UpdatePostRequestType & {
  videoUrl: string;
};

export type UpdatePostRequestUnionType =
  | UpdateTextPostRequestType
  | UpdateVideoPostRequestType;

export type PostResponseType = {
  author: {
    avatar: string;
    firstName: string;
    id: number;
    lastName: string;
    mainInstitution: {
      city: {
        id: number;
        name: string;
      };
      id: number;
      name: string;
    };
  };
  content: string;
  preview: string;
  postType: PostTypeResponseType;
  createdAt: string;
  directions: DirectionResponseType[];
  origins: IOrigin[];
  id: number;
  title: string;
  type: {
    id: number;
    name: string;
  };
  videoUrl?: string;
};

export type ExpertResponseType = {
  avatar: string;
  firstName: string;
  id: number;
  lastAddedPost: {
    id: number;
    title: string;
  };
  lastName: string;
  mainDirection: {
    id: number;
    name: string;
  };
  mainInstitution: {
    city: {
      id: number;
      name: string;
    };
    id: number;
    name: string;
  };
  qualification: string;
};

export type CreateTagRequestType = {
  tag: string;
};

export type TagResponseType = {
  id: number;
  tag: string;
};

export type VersionResponseType = {
  version: string;
};

export type DirectionResponseType = {
  id: number;
  name: string;
  label: string;
  color: string;
};

export type OriginResponseType = {
  id: number;
  name: string;
  parameter: null;
};

export type RegionResponseType = {
  id: number;
  name: string;
};

export type PostTypeResponseType = {
  id: number;
  name: string;
};

export type LoginResponseType = {
  accessToken: string;
  tokenType: string;
};

type GetResponseType<T> = {
  content: T[];
  empty: boolean;
  first: boolean;
  last: boolean;
  number: number;
  numberOfElements: number;
  size: number;
  sort: {
    empty: boolean;
    sorted: boolean;
    unsorted: boolean;
  };
  totalElements: number;
  totalPages: number;
};

export type ExpertsResponseType = GetResponseType<ExpertResponseType>;

export type PostsResponseType = GetResponseType<PostResponseType>;

export type NewestPostsResponseType = GetResponseType<NewestPostResponseType>;

export enum NewestTypeEnum {
  MEDIA,
  TRANSLATION,
  EXPERT_OPINION,
  VIDEO,
}

export type NewestPostResponseType = {
  fieldName: string;
  postDTOS: IPost[];
};

export type GetTagsConfigType = {
  params: {
    value: string;
    limit?: number;
  };
};

type GetConfigType = {
  params: RequestParamsType;
};

export type GetExpertsConfigType = GetConfigType & {
  params: {
    directions?: number[];
    regions?: number[];
  };
};

export type GetPostsConfigType = GetConfigType;

export type RequestParamsType = {
  page?: number;
  size?: number;
  sort?: string[];
  directions?: number[];
  type?: number[];
  types?: number[];
  expert?: number;
  regions?: number[];
  tag?: number[];
  origins?: number[];
  userName?: string;
};
