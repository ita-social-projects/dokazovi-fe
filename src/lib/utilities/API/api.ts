/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable no-param-reassign */
import axios, { AxiosRequestConfig, AxiosResponse } from 'axios';
import qs from 'qs';
import { LocalStorageKeys } from '../../types';
import {
  BASE_URL,
  FACEBOOK_AUTH_URL,
  GOOGLE_AUTH_URL,
  ACCESS_TOKEN_SOCIAL,
} from '../../../apiURL';
import {
  ExpertResponseType,
  GetRegionsType,
  GetResponseType,
  GetTagResponseType,
  GetPostResponseType,
  GetDirectionType,
  GetVersionType,
  PostResponseType,
  PostTagResponseType,
  PostPostRequestType,
  PostLoginResponseType,
  OAuthLoginResponseType,
} from './types';

export const instance = axios.create({
  baseURL: BASE_URL,
});

export type GetConfigType = {
  params: {
    page?: number;
    size?: number;
    sort?: string[];
    direction?: number;
    type?: string[];
    expert?: number;
    regions?: string[];
    tag?: string[];
  };
};

export type GetExpertsConfigType = GetConfigType & {
  params: { directions?: string[]; regions?: string[] };
};

export type GetTagConfigType = {
  params: {
    value: string;
    limit?: number;
  };
};

export type PostTagRequestBodyType = {
  tag: string;
};

instance.interceptors.request.use(
  (config: AxiosRequestConfig) => {
    const jwtToken = localStorage.getItem(LocalStorageKeys.ACCESS_TOKEN);
    if (jwtToken) {
      const header = `Bearer ${jwtToken}`;
      config.headers = { authorization: header };
    }
    return config;
  },
  (error) => {
    Promise.reject(error);
  },
);

instance.interceptors.request.use(
  (config: AxiosRequestConfig) => {
    const jwtToken = ACCESS_TOKEN_SOCIAL;
    if (jwtToken) {
      const header = `Bearer ${jwtToken}`;
      config.headers = { authorization: header };
    }
    return config;
  },
  (error) => {
    Promise.reject(error);
  },
);

const defaultConfig = {
  paramsSerializer: (params: { [key: string]: any }): string => {
    return qs.stringify(params, { arrayFormat: 'repeat' });
  },
};

type PostsTypeType =
  | 'important'
  | 'latest'
  | 'latest-by-direction'
  | 'latest-by-expert';

export const getPosts = (
  postsType: PostsTypeType,
  config?: GetConfigType,
): Promise<AxiosResponse<GetResponseType<PostResponseType>>> => {
  return instance.get<GetResponseType<PostResponseType>>(`/post/${postsType}`, {
    ...defaultConfig,
    ...config,
  });
};

export const getExperts = (
  config?: GetExpertsConfigType,
): Promise<AxiosResponse<GetResponseType<ExpertResponseType>>> => {
  return instance.get(`/user/random-experts`, { ...defaultConfig, ...config });
};

export const getAllExperts = (
  config?: GetExpertsConfigType,
): Promise<AxiosResponse<GetResponseType<ExpertResponseType>>> => {
  return instance.get('user/all-experts', { ...defaultConfig, ...config });
};

export const getPostById = (
  id: number,
): Promise<AxiosResponse<PostResponseType>> => {
  return instance.get(`/post/${id}`);
};

export const getExpertById = (
  id: number,
): Promise<AxiosResponse<ExpertResponseType>> => {
  return instance.get(`/user/${id}`);
};

export const postTag = (
  requestBody: PostTagRequestBodyType,
): Promise<AxiosResponse<PostTagResponseType>> => {
  return instance.post(`/tag`, requestBody);
};

export const getTag = (
  config: GetTagConfigType,
): Promise<AxiosResponse<GetTagResponseType>> => {
  return instance.get(`/tag/find-by-value`, { ...defaultConfig, ...config });
};

export const getVersion = (): Promise<AxiosResponse<GetVersionType>> => {
  return instance.get(`/version`);
};

export const getPostTypes = (): Promise<AxiosResponse<GetPostResponseType>> => {
  return instance.get('post/type');
};

export const getRegions = (): Promise<AxiosResponse<GetRegionsType[]>> => {
  return instance.get(`/region`);
};

export const getDirection = (): Promise<AxiosResponse<GetDirectionType>> => {
  return instance.get(`/direction`);
};

export const postPublishPost = (
  requestBody: PostPostRequestType,
): Promise<AxiosResponse<PostResponseType>> => {
  return instance.post(`/post`, requestBody);
};

export const loginService = (
  email: string,
  password: string,
): Promise<AxiosResponse<PostLoginResponseType>> => {
  return instance.post('/auth/login', { email, password });
};

export const loginFacebook = (): Promise<
  AxiosResponse<OAuthLoginResponseType>
> => {
  return instance.post(FACEBOOK_AUTH_URL);
};

export const loginGoogle = (): Promise<
  AxiosResponse<OAuthLoginResponseType>
> => {
  return instance.post(GOOGLE_AUTH_URL);
};

export const getCurrentUser = (): Promise<
  AxiosResponse<ExpertResponseType>
> => {
  return instance.get('/user/me');
};
