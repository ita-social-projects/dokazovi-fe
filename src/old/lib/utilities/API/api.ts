/* eslint-disable no-param-reassign */
import axios, { AxiosError, AxiosRequestConfig, AxiosResponse } from 'axios';
import qs from 'qs';
import { toast } from 'react-toastify';
import {
  RegionResponseType,
  DirectionResponseType,
  OriginResponseType,
  PostTypeResponseType,
  ExpertResponseType,
  LoginResponseType,
  PostResponseType,
  TagResponseType,
  VersionResponseType,
  CreateTagRequestType,
  ExpertsResponseType,
  PostsResponseType,
  GetTagsConfigType,
  GetExpertsConfigType,
  GetPostsConfigType,
  CreatePostRequestUnionType,
  UpdatePostRequestUnionType,
} from './types';
import { BASE_URL } from '../../../apiURL';
import { getToken } from '../../../provider/AuthProvider/getToken';

export const instance = axios.create({
  baseURL: BASE_URL,
});

instance.interceptors.request.use(
  (config: AxiosRequestConfig) => {
    const jwtToken = getToken();
    if (jwtToken) {
      const header = `Bearer ${jwtToken}`;
      config.headers = { authorization: header };
    }
    return config;
  },
  (error) => {
    Promise.reject(error);
  },
);

instance.interceptors.response.use(undefined, (error: AxiosError) => {
  if (error.message === 'Network Error' && !error.response) {
    toast.error("The server isn't responding...");
  }

  if (!error.response) {
    throw error;
  }

  if (error.response.status === 500) {
    toast.error('A server error occurred...');
  }

  throw error;
});

const defaultConfig = {
  paramsSerializer: (params: { [key: string]: unknown }): string => {
    return qs.stringify(params, { arrayFormat: 'repeat' });
  },
};

type GetPostsRequestType = 'important' | 'latest-all' | 'latest-by-expert';

export const getPosts = async (
  postsRequestType: GetPostsRequestType,
  config?: GetPostsConfigType,
): Promise<AxiosResponse<PostsResponseType>> => {
  return instance.get(`/post/${postsRequestType}`, {
    ...defaultConfig,
    ...config,
  });
};

export const getRandomExperts = async (
  config?: GetExpertsConfigType,
): Promise<AxiosResponse<ExpertsResponseType>> => {
  return instance.get(`/user/random-experts`, { ...defaultConfig, ...config });
};

export const getAllExperts = async (
  config?: GetExpertsConfigType,
): Promise<AxiosResponse<ExpertsResponseType>> => {
  return instance.get('/user/all-experts', { ...defaultConfig, ...config });
};

export const getPostById = async (
  id: number,
): Promise<AxiosResponse<PostResponseType>> => {
  return instance.get(`/post/${id}`);
};

export const getExpertById = async (
  id: number,
): Promise<AxiosResponse<ExpertResponseType>> => {
  return instance.get(`/user/${id}`);
};

export const createTag = async (
  requestBody: CreateTagRequestType,
): Promise<AxiosResponse<TagResponseType>> =>
  instance.post(`/tag`, requestBody);

export const getTagsByValue = async (
  config: GetTagsConfigType,
): Promise<AxiosResponse<TagResponseType[]>> => {
  return instance.get(`/tag/find-by-value`, { ...defaultConfig, ...config });
};

export const getVersion = async (): Promise<
  AxiosResponse<VersionResponseType>
> => {
  return instance.get(`/version`);
};

export const getPostTypes = async (): Promise<
  AxiosResponse<PostTypeResponseType[]>
> => {
  return instance.get('post/type');
};

export const getRegions = async (): Promise<
  AxiosResponse<RegionResponseType[]>
> => {
  return instance.get(`/region`);
};

export const getDirections = async (): Promise<
  AxiosResponse<DirectionResponseType[]>
> => {
  return instance.get(`/direction`);
};

export const getOrigins = async (): Promise<
  AxiosResponse<OriginResponseType[]>
> => {
  return instance.get(`/origin`);
};

export const createPost = async (
  requestBody: CreatePostRequestUnionType,
): Promise<AxiosResponse<PostResponseType>> => {
  return instance.post(`/post`, requestBody);
};

export const updatePost = async (
  requestBody: UpdatePostRequestUnionType,
): Promise<AxiosResponse<PostResponseType>> => {
  return instance.post(`/post`, requestBody);
};

export const login = async (
  email: string,
  password: string,
): Promise<AxiosResponse<LoginResponseType>> => {
  return instance.post('/auth/login', { email, password });
};

export const getCurrentUser = async (): Promise<
  AxiosResponse<ExpertResponseType>
> => {
  return instance.get('/user/me');
};
