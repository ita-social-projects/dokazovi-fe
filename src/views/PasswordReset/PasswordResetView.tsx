import { Container, Typography } from '@material-ui/core';
import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import { useTranslation } from 'react-i18next';
import { BasicAcceptButton, BasicInput } from '../../components/Form';
import { langTokens } from '../../locales/localizationInit';
import { emailValidationObj } from '../../old/lib/components/Users/validationRules';
import { resetPasswordRequest } from '../../old/lib/utilities/API/api';
import { useStyles } from './PasswordResetView.style';

const PasswordResetView = () => {
  const { t } = useTranslation();
  const [submitted, setSubmitted] = useState(false);
  const classes = useStyles();

  // eslint-disable-next-line @typescript-eslint/unbound-method
  const { register, handleSubmit, errors, watch, formState } = useForm<{
    email: string;
  }>({
    mode: 'onTouched',
  });

  const onSubmit = async (inputs: { email: string }) => {
    try {
      await resetPasswordRequest(inputs.email);
    } finally {
      setSubmitted(true);
    }
  };

  const passwordResetView = (
    <form onSubmit={handleSubmit(onSubmit)}>
      <Typography variant="h3">
        {`${t(langTokens.passwordForms.resetPasswordTitle)}:`}
      </Typography>

      <BasicInput
        name="email"
        inputRef={register(emailValidationObj)}
        label="E-mail"
        error={!!errors.email}
        helperText={errors?.email?.message}
      />

      <BasicAcceptButton
        label={t(langTokens.common.acceptChanges)}
        disabled={!formState.isValid || watch('email')?.length === 0}
      />
    </form>
  );

  const submittedView = (
    <Container>
      <Typography variant="h3" component="p" className={classes.centerText}>
        {`${t(langTokens.passwordForms.changePassowrdFromEmail)}!`}
      </Typography>
      <BasicAcceptButton
        label={t(langTokens.passwordForms.repeateAgain)}
        onClick={() => {
          setSubmitted(false);
        }}
      />
    </Container>
  );

  if (submitted) return submittedView;
  return passwordResetView;
};

export default PasswordResetView;
