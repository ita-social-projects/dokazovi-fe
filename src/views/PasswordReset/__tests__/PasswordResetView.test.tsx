import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import PasswordResetView from '../PasswordResetView';
import { resetPasswordRequest } from '../../../old/lib/utilities/API/api';

jest.mock('../../../old/lib/utilities/API/api', () => ({
  resetPasswordRequest: jest.fn(),
}));

describe('PasswordResetView tests', () => {
  it('should PasswordResetView component render', async () => {
    const { asFragment } = render(<PasswordResetView />);
    await waitFor(() => {
      expect(screen.getByTestId('password-reset-view')).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render success view', async () => {
    const { getByTestId } = render(<PasswordResetView />);

    const basicInput = getByTestId('basic-input');

    fireEvent.blur(basicInput);
    userEvent.type(basicInput, 'test@mail.com');

    expect(screen.queryByTestId('submittedContainer')).not.toBeInTheDocument();
    userEvent.click(screen.getByText('Підтвердити зміни'));

    await waitFor(() =>
      expect(screen.getByTestId('submittedContainer')).toBeInTheDocument(),
    );

    expect(resetPasswordRequest).toHaveBeenCalled();
  });

  it('should render previous step', async () => {
    const { getByTestId } = render(<PasswordResetView />);

    const basicInput = getByTestId('basic-input');

    fireEvent.blur(basicInput);
    userEvent.type(basicInput, 'test@mail.com');
    userEvent.click(screen.getByText('Підтвердити зміни'));

    await waitFor(() =>
      expect(screen.getByText('Спробувати ще раз')).toBeInTheDocument(),
    );

    userEvent.click(screen.getByText('Спробувати ще раз'));

    await waitFor(() =>
      expect(screen.queryByText('Спробувати ще раз')).not.toBeInTheDocument(),
    );
  });
});
