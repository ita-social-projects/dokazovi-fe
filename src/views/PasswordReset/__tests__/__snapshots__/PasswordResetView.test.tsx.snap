// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordResetView tests should PasswordResetView component render 1`] = `
<DocumentFragment>
  <form
    data-testid="password-reset-view"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3"
    >
      Відновлення паролю:
    </h3>
    <div
      class="MuiFormControl-root MuiTextField-root BasicInput-basicInput-2 MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink="false"
      >
        E-mail
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          data-testid="basic-input"
          name="email"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-7"
          >
            <span>
              E-mail
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained BasicInput-basicAcceptButton-9 Mui-disabled MuiButton-fullWidth Mui-disabled"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Підтвердити зміни
      </span>
    </button>
  </form>
</DocumentFragment>
`;